cmd/kubeadm/app/componentconfigs/utils.go

```diff
@@ -17,9 +17,29 @@ limitations under the License.
 package componentconfigs
 
 import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/klog"
 )
 
+// UnsupportedConfigVersionError is a special error type returned whenever we encounter too old config version
+type UnsupportedConfigVersionError struct {
+	// OldVersion is the config version that is causing the problem
+	OldVersion schema.GroupVersion
+
+	// CurrentVersion describes the natively supported config version
+	CurrentVersion schema.GroupVersion
+
+	// Document points to the YAML/JSON document that caused the problem
+	Document []byte
+}
+
+// Error implements the standard Go lang error interface for UnsupportedConfigVersionError
+func (err *UnsupportedConfigVersionError) Error() string {
+	return fmt.Sprintf("unsupported apiVersion %q, you may have to do manual conversion to %q and run kubeadm again", err.OldVersion, err.CurrentVersion)
```

Comment:
i guess if a component ends up with multiple Kinds for a GroupVersion (future kube-proxy?) one way to indicate what Kind caused the error would be to look at the Document bytes.

Reply:
Yep. That's why the yaml slice is there. However, the error message would become too complicated if that was to be dumped here. Instead, those document bytes can be dumped on a higher level upon recognizing this error and with a better UX in mind.